Remove usage of legacy scheme paths on RedoxOS

bjorn3 opened the pull request:
The `name:/path` path syntax is getting phased out[^1] in favor of `/scheme/name/path`. Also using `null:` is no longer necessary as `/dev/null` is available on Redox OS too.

[^1]: https://gitlab.redox-os.org/redox-os/rfcs/-/blob/master/text/0006-scheme-path.md

cc @jackpot51

bjorn3 committed:
Remove usage of legacy scheme paths on RedoxOS

The name:/path path syntax is getting phased out in favor of
/scheme/name/path. Also using null: is no longer necessary as /dev/null
is available on Redox OS too.

bors commented:
:pushpin: Commit 83ee034d037d2386085ce44bddad286c24e4ed6c has been approved by `Noratrieb`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Noratrieb 83ee034d037d2386085ce44bddad286c24e4ed6c -->
<!-- homu: {"type":"Approved","sha":"83ee034d037d2386085ce44bddad286c24e4ed6c","approver":"Noratrieb","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bjorn3 committed:
Remove has_redox_scheme

Redox OS is moving away from name:/path style paths to /scheme/name/path
style paths which are already handled correctly without has_redox_scheme.

bjorn3 commented:
Added another commit.

@bors r-
@rustbot ready

Noratrieb commented:
@bors r+

bors commented:
:pushpin: Commit 43499bfe2ea2b37f98d2bbfcdda05930b20c5591 has been approved by `Noratrieb`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Noratrieb 43499bfe2ea2b37f98d2bbfcdda05930b20c5591 -->
<!-- homu: {"type":"Approved","sha":"43499bfe2ea2b37f98d2bbfcdda05930b20c5591","approver":"Noratrieb","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
